### PR TITLE
Storing CodeProblem.type values separately, also renaming error field to description

### PR DESCRIPTION
**Pull request by isoos**

### CODE REVIEW

File: lib/src/fitness.dart [hunk 1]
```diff
@@ -77,9 +77,9 @@ Future<Fitness> calcFitness(
 
   if (fileAnalyzerItems != null) {
     for (var item in fileAnalyzerItems) {
-      if (item.type.startsWith('INFO')) {
+      if (item.severity.startsWith('INFO')) {
```
Comment (kevmoo) [inline lib/src/fitness.dart]: can now be `==` now

File: lib/src/fitness.dart [hunk 2]
```diff
@@ -77,9 +77,9 @@ Future<Fitness> calcFitness(
...
         penalties += hintPoints;
-      } else if (item.type.startsWith('WARN')) {
+      } else if (item.severity.startsWith('WARN')) {
```
Comment (kevmoo) [inline lib/src/fitness.dart]: Ditto - can now be `==` now

Comment (isoos) [reply lib/src/fitness.dart]: is it 'WARN' or 'WARNING'? I wasn't sure, hence the startsWith

**Comment by kevmoo** [comment]
WARNING – good call!

**Pull request merged**